src/main/java/de/tum/in/www1/artemis/service/ExamService.java

```diff
@@ -201,8 +201,7 @@ public void delete(Long examId) {
      * @return return ExamScoresDTO with students, scores and exerciseGroups for exam
      */
     public ExamScoresDTO getExamScore(Long examId) {
-        Exam exam = examRepository.findWithRegisteredUsersAndExerciseGroupsAndExercisesById(examId)
-                .orElseThrow(() -> new EntityNotFoundException("Exam with id: \"" + examId + "\" does not exist"));
+        Exam exam = examRepository.findOneWithEagerExercisesGroupsAndStudentExams(examId);
```

Comment:
please double check that this call also works without any issues when there are many (i.e. >1000) student exams. We previously had some issues when too many entities in the database were fetched. I assume there is an edge case when it comes to the relationships between student exam and some other entities (we don't know the exact reason yet).

It is more safe to fetch 2 times:
1) Exam with exercise groups and exercises
2) List of all students exams for one particular exam (without any other entities).

This should also be faster in terms of performance and memory consumption, because the entity graph will be much simpler!

Can you please change it accordingly?

Reply:
See https://github.com/ls1intum/Artemis/pull/1982/commits/1ba0f668b248510539b168f45d42f27fb077086e